### PR TITLE
ENH: Add SeparableSum(func, n), closes #685

### LINKED ISSUE #685: Add syntax SeparableSum(func, n)
Our other productspace operators and classes support the syntax

```
>>> odl.ProductSpace(space, n)
>>> odl.DiagonalOperator(space, n)
# etc
```

`SeparableSum` should support this as well.

### PR DESCRIPTION
**Pull request by adler-j**

### CODE REVIEW

File: odl/solvers/functional/default_functionals.py [hunk 1]
```diff
@@ -1140,7 +1141,27 @@ def __init__(self, *functionals):
         ----------
         functional1, ..., functionalN : `Functional`
             The functionals in the sum.
+            Can also be given as ``space, n`` with ``n`` integer,
+            in which case the functional is repeated ``n`` times.
+
+        Examples
+        --------
+        Create functional ``f([x1, x2]) = ||x1||_1 + ||x2||_2``
```
Comment (kohr-h) [inline odl/solvers/functional/default_functionals.py]: colon

File: odl/solvers/functional/default_functionals.py [hunk 2]
```diff
@@ -1140,7 +1141,27 @@ def __init__(self, *functionals):
...
+
+        >>> space = odl.rn(3)
+        >>> l1 = odl.solvers.L1Norm(space)
+        >>> l2 = odl.solvers.L2Norm(space)
+        >>> f_sum = odl.solvers.SeparableSum(l1, l2)
+
+        Create functional ``f([x1, ... ,xn]) = \sum_i ||xi||_1``
```
Comment (kohr-h) [inline odl/solvers/functional/default_functionals.py]: colon

**Comment by adler-j** [comment]
Fixed comments and added a proper test. Merge?

**Comment by kohr-h** [comment]
Nice, go ahead.

**Pull request merged**